src/utils/CdnUtils.ts

```diff
@@ -0,0 +1,55 @@
+import axios from "axios";
+import R2Error from "../model/errors/R2Error";
+
+const CDNS = [
+    "gcdn.thunderstore.io",
+    "hcdn-1.hcdn.thunderstore.io"
+]
+const TEST_FILE = "healthz";
+
+const canReachCdn = async (domain: string) => {
+    const url = `https://${domain}/${TEST_FILE}`;
+    const headers = {
+        "Cache-Control": "no-cache",
+        "Pragma": "no-cache",
+        "Expires": "0",
+    };
+    const params = {"disableCache": new Date().getTime()}
+    let res;
+
+    try {
+        res = await axios.get(url, {headers, params});
+    } catch (e) {
+        return false;
+    }
+
+    return res.status === 200;
+}
+
+export const getReachableCdn = async () => {
+    for await (const cdn of CDNS) {
+        if (await canReachCdn(cdn)) {
+            return cdn;
+        }
+    };
+
+    throw new R2Error(
+        "Can't reach content delivery networks",
+        `Both Thunderstore CDNs seem to be currently unreachable from
```

Comment:
Would replace `Both` with `All`, not a blocker

Reply:
Fixed, was leftover from the original implementation that supported just one alternative mirror.